lib/core/src/Cardano/Wallet/Primitive/Model.hs

```diff
@@ -376,7 +377,8 @@ prefilterBlock b u0 = runState $ do
         ourU <- state $ utxoOurs tx
         let ourIns = Set.fromList (inputs tx) `Set.intersection` dom (u <> ourU)
         let u' = (u <> ourU) `excluding` ourIns
-        let received = fromIntegral @_ @Integer $ balance ourU
+        let withdrawalAmt = sum $ map (fromIntegral . getCoin) $ Map.elems $ withdrawals tx
+        let received = fromIntegral @_ @Integer $ (withdrawalAmt + balance ourU)
```

Comment:
withdrawals here also need to be filtered based on whether they're ours or not. Unfortunately, that's the same bits of logic in two different places.

Reply:
addressed in https://github.com/input-output-hk/cardano-wallet/pull/1942/commits/f200ae0bb1c9d9e7184a3a073f0b85bb6524bf1a